browser/ui/webui/brave_web_ui_controller_factory.cc

```diff
@@ -195,6 +196,12 @@ WebUIController* NewWebUI(WebUI* web_ui, const GURL& url) {
 WebUIFactoryFunction GetWebUIFactoryFunction(WebUI* web_ui,
                                              Profile* profile,
                                              const GURL& url) {
+  // This will get called a lot to check all URLs, so do a quick check of other
+  // schemes to filter out most URLs.
+  if (!content::HasWebUIScheme(url)) {
```

Comment:
@simonhong this includes devtools and untrusted. are there any problems if for instance a user goes to `chrome-untrusted://settings`? it seems maybe the checks for host_piece should check for the full origin below.

Reply:
Seems to be OK - but I have a follow up. Will share soon